### PR TITLE
PR #58 is not compatible with previous version...

### PR DESCRIPTION
**Pull request by k-okada**
we may need this for naoqi_driver to perform in same namespace,,, #58

### CODE REVIEW

**Comment by vrabaud** [comment]
@suryaambrose , any input ?

File: src/ros_env.hpp [hunk 1]
```diff
@@ -89,7 +89,8 @@ static void setMasterURI( const std::string& uri, const std::string& network_int
   remap["__master"] = uri;
   remap["__ip"] = ::naoqi::ros_env::getROSIP(network_interface);
   // init ros without a sigint-handler in order to shutdown correctly by naoqi
-  ros::init( remap, ::naoqi::ros_env::getPrefix().empty()?std::string("naoqi_driver_node"):(::naoqi::ros_env::getPrefix()) , ros::init_options::NoSigintHandler );
+  const char* ns_env = std::getenv("ROS_NAMESPACE");
+  ros::init( remap, (ns_env==NULL)?(std::string("naoqi_driver_node")):(::naoqi::ros_env::getPrefix()) , ros::init_options::NoSigintHandler );
```
Comment (suryaambrose) [inline src/ros_env.hpp]: So what happens if I do not define a ROS_NAMESPACE variable and do not define a prefix either ? Isn't it the same as before (no namespace found) ?

**Comment by suryaambrose** [comment]
I don't really understand what you are trying to fix here, what is the problem actually ?

**Comment by k-okada** [comment]
1. The original problem was reported by https://github.com/ros-naoqi/naoqi_driver/pull/44, which could not 
set proper node name when no ROS namespace is found. (which usually not happens when we used nao or pepper robot)
2. I thouhg this was fixed by https://github.com/ros-naoqi/naoqi_driver/pull/58/files, but it is not correct. this will set ros node name as `<namespace>/naoqi_driver_node`, because `::naoqi::ros_env::getPrefix()` is already initialized at  https://github.com/k-okada/naoqi_driver.cpp/blob/fix_58/src/naoqi_driver.cpp#L126

**Comment by suryaambrose** [comment]
OK thanks for the explanation. 
I wanted to test that today, but I really don't have time. However it seems good to me, so if it works as expected for you, go.

**Pull request merged**